### ISSUE TITLE
Can't add homebridge to Home app

### ISSUE THREAD

**Issue by kaganae** [nfarina/homebridge#2092]
Got Homebridge installed to use the PowerView plugin. All looks fine on my Mac (running Mojave) but then get "Couldn't add homebridge-CAD8" error in Home app. I have tried removing the persist directory and changing the username and pin in config.json, but no luck. Any ideas?

### Versions

- **Homebridge**: `homebridge --version`: 0.4.45
- **Node**:       `node --version` v11.5.0
- **OS**:         MacOS Mojave 10.14.2

Config.json:

{
        "bridge": {
        "name": "homebridge",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-152"
    },

    "platforms": [
        {
            "platform" : "PowerView",
            "host" : "192.168.10.185"
        }
    ]
}

At startup: 

[12/19/2018, 11:00:04 PM] Loaded config.json with 0 accessories and 1 platforms.
[12/19/2018, 11:00:04 PM] ---
[12/19/2018, 11:00:04 PM] Loaded plugin: homebridge-powerview
[12/19/2018, 11:00:04 PM] Registering platform 'homebridge-powerview.PowerView'
[12/19/2018, 11:00:04 PM] ---
[12/19/2018, 11:00:04 PM] Loading 1 platforms...
[12/19/2018, 11:00:04 PM] [PowerView] Initializing PowerView platform...
[12/19/2018, 11:00:04 PM] [PowerView] PowerView init
[12/19/2018, 11:00:04 PM] [PowerView] Cached shade 39645: Alex
[12/19/2018, 11:00:04 PM] [PowerView] Cached shade 38787: Living Room Left
[12/19/2018, 11:00:04 PM] [PowerView] Cached shade 10400: Master Bathroom
[12/19/2018, 11:00:04 PM] [PowerView] Cached shade 32918: Basement Left
[12/19/2018, 11:00:04 PM] [PowerView] Cached shade 43815: Living Room Right
[12/19/2018, 11:00:04 PM] [PowerView] Cached shade 56498: Dining Room
[12/19/2018, 11:00:04 PM] [PowerView] Cached shade 51506: Basement Right
Setup Payload:
X-HM://0023ISYWWHKAA
Scan this code with your HomeKit app on your iOS device to pair with Homebridge:
                                                      
                                                  
Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 031-45-152 │     
    └────────────┘     
                       
[12/19/2018, 11:00:04 PM] Homebridge is running on port 51826.
[12/19/2018, 11:00:04 PM] [PowerView] Updating shade 39645: Alex
[12/19/2018, 11:00:04 PM] [PowerView] *** Shade 39645 has unknown type 6, assuming roller ***
[12/19/2018, 11:00:04 PM] [PowerView] Updating shade 38787: Living Room Left
[12/19/2018, 11:00:04 PM] [PowerView] *** Shade 38787 has unknown type 6, assuming roller ***
[12/19/2018, 11:00:04 PM] [PowerView] Set for 38787 { positions: { posKind1: 1, position1: 0 } }
[12/19/2018, 11:00:04 PM] [PowerView] Updating shade 10400: Master Bathroom
[12/19/2018, 11:00:04 PM] [PowerView] *** Shade 10400 has unknown type 6, assuming roller ***
[12/19/2018, 11:00:04 PM] [PowerView] Updating shade 32918: Basement Left
[12/19/2018, 11:00:04 PM] [PowerView] *** Shade 32918 has unknown type 6, assuming roller ***
[12/19/2018, 11:00:04 PM] [PowerView] Set for 32918 { positions: { posKind1: 1, position1: 0 } }
[12/19/2018, 11:00:04 PM] [PowerView] Updating shade 43815: Living Room Right
[12/19/2018, 11:00:04 PM] [PowerView] *** Shade 43815 has unknown type 6, assuming roller ***
[12/19/2018, 11:00:04 PM] [PowerView] Set for 43815 { positions: { posKind1: 1, position1: 0 } }
[12/19/2018, 11:00:04 PM] [PowerView] Updating shade 56498: Dining Room
[12/19/2018, 11:00:04 PM] [PowerView] *** Shade 56498 has unknown type 6, assuming roller ***
[12/19/2018, 11:00:04 PM] [PowerView] Set for 56498 { positions: { posKind1: 1, position1: 0 } }
[12/19/2018, 11:00:04 PM] [PowerView] Updating shade 51506: Basement Right
[12/19/2018, 11:00:04 PM] [PowerView] *** Shade 51506 has unknown type 6, assuming roller ***
[12/19/2018, 11:00:04 PM] [PowerView] Set for 51506 { positions: { posKind1: 1, position1: 0 } }

**Comment by luishasmail** [comment]
Has this setup worked in the past or is this your first time setting it up?

**Comment by kaganae** [comment]
It’s the first time.

**Comment by luishasmail** [comment]
There's a number of things that I always check for the first time. First I check to make sure that my environment is sane. I use this docker container, but you can check the setup steps in this docker file to make sure everything is there: https://github.com/oznu/docker-homebridge/blob/master/Dockerfile If you don't know docker, I would double check the steps in the README on the repo.

Another thing I keep an eye out for is the basic sanity checking: Booting up with no plugins, making sure my devices are on the same network, deleting persist and any instances on the home app, making sure I changed the pin and username from the default config, etc.

If you give me some more detailed steps I can try to reproduce this on my Mac and see if I can help debug it.

**Comment by kaganae** [comment]
Thanks. I am not a docker person, struggling to figure that one out. Basically, I followed the instructions to install Homebridge, execute it, then I use the Home app to add the accessory but it says it can’t add because “Home couldn’t connect to this accessory”. I have tried connecting my iMac that is running it over both Ethernet and Wifi connections. I have also removed the plugin to just have a vanilla homebridge running and changed the username and pin, but always get the same results.

**Comment by luishasmail** [comment]
I attempted to reproduce this using a VM running macOS and everything worked fine 

Here are the versions for the tooling I used:

| Item | Version |
|------------|----------|
| node       | v10.14.2 |
| npm        | 6.4.1    |
| homebridge | 0.4.45   |
| macOS      | 10.14.2  |
| iOS        | 12.1.1   |

Here is my config:

```json
{
  "bridge": {
    "name": "Foo",
    "username": "AA:11:33:EE:EE:33",
    "port": 51826,
    "pin": "765-99-211"
  },
  "description": "none",
  "accessories": [],
  "platforms": []
}
```

Here's a rough outline of how I set it up:

```bash
# Install homebrew

/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"

# Install nvm

brew update
brew install nvm

mkdir ~/.nvm

# Update bash profile according to nvm install 

# install nodejs

nvm install 10

npm install -g homebridge

homebridge

touch ~/.homebridge/config.json

homebridge
```

I also tried this using node 11 and it still worked. Are there any discrepancies between how you are setup? Have you checked to make sure that the machine running homebridge is reachable from your phone?

**Comment by kaganae** [comment]
Thanks for al of the help. I deleted everything and did a fresh install, following your steps and config file, with the same results (unable to add Homebridge to the Home app), unfortunately. 

I’m not sure if you are familiar with Home Assistant, but that is another open source Home Automation solution that is available. I have installed that on the same iMac and it works fine (although I don’t think the plugins are as robust, so would like to make Homebridge work). It doesn’t seem to be a network issue.

**Comment by dwh17** [comment]
I've recently started having the same issue as well.  I'm running a Pi3 with the lastest Raspbian and everything updated.  Previously, Homebridge worked fine with the Wink3 plugin, but then I did a fresh OS install so I could also install NextCloud on the Pi.  After installing Homebridge with persistence and the Wink3 plugin, Homebridge initially would not connect locally to the Wink hub, but a re-login to the Wink developer site fixed that.  Now this is what the log shows when I start Homebridge:

[2019-2-10 12:45:49] Loaded config.json with 0 accessories and 1 platforms.
[2019-2-10 12:45:49] ---
[2019-2-10 12:45:51] Loaded plugin: homebridge-wink3
[2019-2-10 12:45:51] Registering platform 'homebridge-wink.Wink'
[2019-2-10 12:45:51] ---
[2019-2-10 12:45:51] Loading 1 platforms...
[2019-2-10 12:45:51] [Wink] Initializing Wink platform...
[2019-2-10 12:45:52] [Wink] Loaded from cache: GE Link Bulb (light_bulb/696140)
[2019-2-10 12:45:52] [Wink] Loaded from cache: GE Link Bulb (light_bulb/696141)
Setup Payload:
X-HM://0023ISYWYOPAN
Scan this code with your HomeKit app on your iOS device to pair with Homebridge:
Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ xxx-xx-xxx
    └────────────┘     
                       
[2019-2-10 12:45:52] Homebridge is running on port 51826.
[2019-2-10 12:45:52] [Wink] Refreshing devices...
[2019-2-10 12:45:53] [Wink] Devices refreshed
[2019-2-10 12:45:53] [Wink] Checking if hub is reachable (LR and Den Hub, 192.168.0.y)...
[2019-2-10 12:45:53] [Wink] Hub is reachable locally (LR and Den Hub, 192.168.0.y)
[2019-2-10 12:45:53] [Wink] Authenticated with local hub (LR and Den Hub, 192.168.0.y)

The Hub shows up when I try to use "Add accessory" to add it to the Home app (as "Homebridge-210D," which I don't recall naming it), but I then get "Couldn't add Homebridge-21D, Home couldn't connect to this accessory."

I've reinstalled and reset everything,  Any ideas would be appreciated. Thanks.
